service/src/models/archaeologist.ts

```diff
@@ -124,24 +125,41 @@ export class Archaeologist {
         await pipe(
           stream,
           async (source) => {
-            for await (const data of source) {
-              const jsonData = JSON.parse(new TextDecoder().decode(data));
-
-              const txId = jsonData.arweaveTxId;
-              const unencryptedShardHash = jsonData.unencryptedShardHash;
-
-              const isValidShard = await fetchAndValidateArweaveShard(txId, unencryptedShardHash, this.web3Interface.encryptionWallet.publicKey);
+            const rejectTransaction = () => streamToBrowser(JSON.stringify({ signature: '' }))
+            const signOff = (signature: string) => streamToBrowser(JSON.stringify({ signature }));
+            const emitError = (error: any) => streamToBrowser(JSON.stringify({ error }));
 
-              if (isValidShard) {
-                const msg = ethers.utils.solidityPack(
-                  ['string', 'string', 'string'],
-                  [txId, unencryptedShardHash, this.web3Interface.ethWallet.address]
-                )
-                const signature = await this.web3Interface.encryptionWallet.signMessage(msg);
-
-                streamToBrowser(signature);
-              } else {
-                streamToBrowser('0');
+            for await (const data of source) {
+              try {
+                const jsonData: {
+                  arweaveTxId: string,
+                  unencryptedShardDoubleHash: string,
+                  maxRewrapInterval: number,
+                } = JSON.parse(new TextDecoder().decode(data));
```

Comment:
If you wanted to get around having to re-assign variables from this json object (or call `jsonData.x`), could define an interface outside of this function somewhere and use it, like:

```
interface SarcoDataFromEmbalmerToValidate {
  arweaveTxId: string;
  unencryptedShardDoubleHash: string;
  maxRewrapInterval;
}
...
const {
    arweaveTxId,
    unencryptedShardDoubleHash,
    maxRewrapInterval
  }: SarcoDataFromEmbalmerToValidate = JSON.parse(new TextDecoder().decode(data));
```

Reply:
Oh sweet, gotta love destructuring. Thanks for the reminder!